static/js/public/snap-details/channelMap.js

```diff
@@ -69,31 +69,52 @@ function setArchitecture(arch, packageName, channelMapData) {
 
   const tracks = Object.keys(channelMapData[arch]);
 
-  // sort tracks alphabetically with 'latest' always first
-  tracks.sort((a, b) => {
-    if (a === LATEST) {
-      return -1;
-    }
-    if (b === LATEST) {
-      return 1;
+  // split tracks into strings and numbers
+  let numberTracks = [];
+  let stringTracks = [];
+  tracks.forEach(track => {
+    // numbers are defined by any string starting any of the following patterns:
+    //   just a number – 1,2,3,4,
+    //   numbers on the left in a pattern – 2018.3 , 1.1, 1.1.23 ...
+    //   or numbers on the left with strings at the end – 1.1-hotfix
+    if (isNaN(parseInt(track.substr(0, 1)))) {
+      stringTracks.push(track);
+    } else {
+      numberTracks.push(track);
     }
-    return a <= b ? -1 : 1;
   });
 
-  // by default take first track (which should be 'latest')
-  const track = tracks[0];
+  // Ignore case
+  stringTracks.sort(function (a, b) {
+    return a.toLowerCase().localeCompare(b.toLowerCase());
+  });
+
+  const latestIndex = stringTracks.indexOf(LATEST);
+  stringTracks.splice(latestIndex, 1);
+  stringTracks = [LATEST].concat(stringTracks);
+
+  // Sort numbers (that are actually strings)
+  numberTracks.sort((a, b) => {
+    return b.localeCompare(a, undefined, { numeric: true, sensitivity: 'base' });
+  });
+
+  // Join the arrays together again
+  const sortedTracks = stringTracks.concat(numberTracks);
+
+  // by default take first track (which should be 'latest') unless it's node
+  const track = packageName === 'node' ? sortedTracks[sortedTracks.indexOf('10')] : sortedTracks[0];
```

Comment:
I know that node is a special case, but adding special treatment in different places (first backend, now channel map) will make it much more difficult to track them down later (especially that this is added in totally unrelated PR).

Reply:
The issue that `node` has 'latest' selected by default instead of '10' is separate from sorting, and maybe at least it would be good to solve it separately (for it to be found easier later). Or ideally to solve it in a different way that doesn't require hardcoding it into JS. Would be good if backend could give us a current track name, so we would default to this instead of 'latest'.

Reply:
Forgot I added that. Did it as a little test to see if it would work as expected. Removed.